kolibri2zim/scraper.py

```diff
@@ -1059,13 +1034,58 @@ def add_custom_about_and_css(self):
             if not user_provided_file:
                 content = ""
             else:
-                with open(user_provided_file) as fh:
-                    content = fh.read()
+                content = user_provided_file.read_bytes()
         # otherwise, create a blank one
         else:
             content = ""
 
-        self.creator.add_item_for(
-            "custom.css", content=content.encode(), mimetype="text/css"
-        )
+        self.creator.add_item_for("custom.css", content=content, mimetype="text/css")
         logger.debug("Added about page and custom CSS")
+
+    def ensure_js_deps_are_present(self):
+        for js_deps_file in [
```

Comment:
Make a list of path that are pointing either to file or folder ; turn them into Path (no `os.path`!) and check only once instead of duplicating all this code.
I don't think file vs folder matters here but if you do, just switch message based on `.suffix`